nnunetv2/imageio/nibabel_reader_writer.py

```diff
@@ -78,14 +78,13 @@ def read_images(self, image_fnames: Union[List[str], Tuple[str, ...]]) -> Tuple[
             print(image_fnames)
             raise RuntimeError()
 
-        stacked_images = np.vstack(images)
         dict = {
             'nibabel_stuff': {
                 'original_affine': original_affines[0],
             },
             'spacing': spacings_for_nnunet[0]
         }
-        return stacked_images.astype(np.float32), dict
+        return np.vstack(images, dtype=np.float32, casting='unsafe'), dict
```

Comment:
same question as above

Reply:
Casting from integers to floating point can only be done using 'unsafe' casting.